### DISCUSSION TITLE
Passing Control as CommandParameter with Compiled Bindings

### DISCUSSION THREAD

**Discussion by robloo** [AvaloniaUI/Avalonia#14353] [Q&A] [answered]
I have the below XAML. Null is passed for the CommandParameter with a compiled binding. If I switch to a ReflectionBinding the control is passed to the command as expected.

I realize this isn't ideal code but there are some cases where just passing the control reference to the view model temporarily is quicker and easier than alternatives (reference isn't stored and it's only used in the command). My question: is it expected for this to not work under compiled bindings but work under reflection bindings? Or is this another case where compiled bindings are incomplete?

The example XAML uses FluentAvalonia controls quite extensively but I don't think that influences behavior here.

```xaml
<fac:CommandBarToggleButton
  x:Name="ListViewToggleButton"
  Command="{Binding SetView}"
  CommandParameter="{ReflectionBinding ElementName=ListViewToggleButton}">
  <fac:CommandBarToggleButton.IconSource>
    <fac:FontIconSource
    Glyph="&#xF4E1;"
    FontFamily="{DynamicResource FluentUISystemRegular}"
    FontSize="{DynamicResource AppSymbolFontSize}" />
  </fac:CommandBarToggleButton.IconSource>
</fac:CommandBarToggleButton>
```

**Comment by workgroupengineering** [comment, marked as answer]
Why do you use `{Reflection Binding ElementName=ListView Toggle Button}` instead of `{Binding #ListViewToggleButton}` or `{Biding #self}`?

Try with those.

**Reply by robloo** [reply]
I will try those thanks, this is ported code from WinUI and I have vastly more experience with the WPF syntax.

**Reply by timunie** [reply]
Binding $self, not #self.

**Reply by robloo** [reply]
Both recommended Avalonia-specific syntaxes mentioned above work. However, I still wonder if this should be filed as an issue since compiled and reflection bindings really shouldn't be different like this. I might start a catch-all issue tracker for differences.
